spring-cloud-azure-config/src/main/java/com/microsoft/azure/config/ConfigHttpClient.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.config;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.net.URISyntaxException;
+import java.nio.charset.Charset;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.codec.digest.HmacAlgorithms;
+import org.apache.commons.codec.digest.HmacUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpEntityEnclosingRequest;
+import org.apache.http.HttpRequest;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+import static org.apache.commons.codec.digest.HmacAlgorithms.HMAC_SHA_256;
+import static org.apache.commons.codec.digest.MessageDigestAlgorithms.SHA_256;
+
+/**
+ * Util class to execute http request, before sending http request, valid request headers
+ * will be added for each request based on given credential ID and secret.
+ *
+ * How to use:
+ * <p>
+ * HttpGet httpGet = new HttpGet("https://my-config-store.azconfig.io/keys");
+ * CloseableHttpResponse response = ConfigHttpClient.execute(httpGet, "my-credential",
+ * "my-secret");
+ * <p/>
+ */
+@Slf4j
+public class ConfigHttpClient {
+    private static final String DATE_FORMAT = "EEE, d MMM yyyy hh:mm:ss z";
+
+    private static final SimpleDateFormat GMT_DATE_FORMAT = new SimpleDateFormat(DATE_FORMAT);
+
+    static {
+        GMT_DATE_FORMAT.setTimeZone(TimeZone.getTimeZone("GMT"));
+    }
+
+    public static CloseableHttpResponse execute(HttpUriRequest request, String credential, String secret)
+            throws IOException, URISyntaxException {
+        Map<String, String> authHeaders = buildRequestHeaders(request, new Date(), credential, secret);
+        authHeaders.forEach(request::setHeader);
+
+        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+            return httpClient.execute(request);
+        }
+    }
+
+    private static Map<String, String> buildRequestHeaders(HttpUriRequest request, Date date, String credential,
+            String secret) throws URISyntaxException, IOException {
+        String requestTime = GMT_DATE_FORMAT.format(date);
+
+        String content = "";
```

Comment:
Suggest to extract another method only for build signature. This will make this method easier to understand.

Reply:
Sure, will extract to another method.